scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/ModelValidator.java

```diff
@@ -0,0 +1,90 @@
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.common.job.factories.spi.model.validator;
+
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.exceptions.ValidationException;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.BooleanParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.DateTimeParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.DoubleParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.FloatParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.IntegerParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.ListParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.LongParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.ModelFromURLParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.RegexpParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.ShortParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.URIParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.URLParserValidator;
+
+import com.google.common.base.Strings;
+
+
+public class ModelValidator implements Validator<String> {
+
+    private String model;
+
+    public ModelValidator(String model) {
+        if (Strings.isNullOrEmpty(model)) {
+            throw new IllegalArgumentException("Model cannot be empty");
+        }
+        this.model = model.trim();
+    }
+
+    @Override
+    public String validate(String parameterValue) throws ValidationException {
+        String uppercaseModel = model.toUpperCase();
+        try {
+            if (uppercaseModel.startsWith(IntegerParserValidator.INTEGER_TYPE)) {
```

Comment:
Why `startsWith` and not `equals`?

If `equals`, you could define a map to look up the validator to use, thus avoiding a lot of branching, increasing readability and maintenance.